### PR TITLE
detect volume we didn't created and ask user to explicitely mark them as external

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
On `compose up` we search for existing volume by name, as for legacy and usability reasons this one is determined by project name. Doing so we risk to use an existing volume that was not created by Docker Compose or was created by another compose project.

This PR adds a check that volume doesn't match expected project, and tells user to make this explicitly `external: true` if this is really what he expects.

**Related issue**
see https://github.com/docker/compose/issues/8850

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**

### CODE REVIEW

File: pkg/compose/create.go [hunk 1]
```diff
@@ -1086,27 +1086,42 @@ func (s *composeService) removeNetwork(ctx context.Context, networkID string, ne
...
+	}
+
+	// Volume exists with name, but let's double check this is the expected one
+	// (better safe than sorry when it comes to user's data)
+	p, ok := inspected.Labels[api.ProjectLabel]
+	if !ok {
+		return fmt.Errorf("volume %q alrady exists but was not created by Docker Compose. Use `external: true` to use an existing volume", volume.Name)
+	}
+	if p != project {
+		return fmt.Errorf("volume %q alrady exists but was not created for project %q. Use `external: true` to use an existing volume", volume.Name, p)
```
Comment (glours) [inline pkg/compose/create.go:1106]:
```suggestion
		return fmt.Errorf("volume %q already exists but was not created for project %q. Use `external: true` to use an existing volume", volume.Name, p)
```

File: pkg/compose/create.go [hunk 2]
```diff
@@ -1086,27 +1086,42 @@ func (s *composeService) removeNetwork(ctx context.Context, networkID string, ne
...
-		w.Event(progress.CreatedEvent(eventName))
+		err := s.createVolume(ctx, volume)
+		return err
+	}
+
+	// Volume exists with name, but let's double check this is the expected one
+	// (better safe than sorry when it comes to user's data)
+	p, ok := inspected.Labels[api.ProjectLabel]
+	if !ok {
+		return fmt.Errorf("volume %q alrady exists but was not created by Docker Compose. Use `external: true` to use an existing volume", volume.Name)
```
Comment (glours) [inline pkg/compose/create.go:1103]:
```suggestion
		return fmt.Errorf("volume %q already exists but was not created by Docker Compose. Use `external: true` to use an existing volume", volume.Name)
```

**Review by glours — APPROVED** [review]
Only 2 typo to fix before merging

**Comment by glours** [comment]
Ho  I didn't see that the CI was red 😬

**Comment by ndeloof** [comment]
yes, CI suffers for flakiness, I use to re-run mostly on all pull-request to get a successful build
(I wonder we could enable auto-retry on tests :P )

**Comment by glours** [comment]
Or we can spend some time to figure why the tests are flaky 😛

**Comment by MillerCM** [comment]
A request: Since the users of docker are anywhere and everywhere on the gender spectrum, please make an effort to be inclusive with your phrasing. The phrase "if this is really what he expects" is unnecessarily gendering the user as male. It's more accurate to represent your entire user base by repeating "the user" or using a neutral pronoun (historically, "they").

Thank you for your work!

**Pull request merged**